docs/developer/events.md

```diff
@@ -41,13 +41,18 @@ public class ExampleEventSubscriber implements EventSubscriber {
 }
 ```
 
-All the events are subclass of the `Event` abstract class so to filter events it's enough to check the type at runtime.
+To filter events, the payload classes of the events can be used. There are 5 "Intermediate superclasses" (AssetEventPayload, 
+ContractDefinitionEventPayload, ContractNegotiationEventPayload, PolicyDefinitionEventPayload, TransferProcessEventPayload) of 
+which one payload class inherits in each case. These intermediate superclasses inherit again from the class EventPayload. Thus, 
+in addition to filtering on a specific event, such as TransferProcessCompleted, it is possible to react to a group of events 
+that generally have to do with Assets, ContractDefinition, ContractNegotiation, PolicyDefinition and TransferProcess.
+
 ```java
 public class ExampleEventSubscriber implements EventSubscriber {
     
     public void on(Event event) {
-        if (event instanceof AssertCreated) {
-            // react only to AssertCreated events
+        if (event.getPayload() instanceof TransferProcessEventPayload) {
```

Comment:
This actually can be done in both ways, filtering by event class type or by payload as well, I would keep both the examples

Reply:
Added it back. Hope its fine.